### ISSUE TITLE
Ray Tune Keras Integration (ray.tune.integration.keras.TuneReportCheckpointCallback) not working correctly out of the box

### ISSUE THREAD

**Issue by hvavhp** [ray-project/ray#30236]
### What happened + What you expected to happen

Hi, I'm not sure if this is a bug or a feature, but since I had to spent quite sometimes to get around this, I'd like to share, at least for anyone who got stuck. The issue here is with the checkpoint integration callback for Keras `ray.tune.integration.keras.TuneReportCheckpointCallback`. I will not provide any reproduction script because it is rather straightforward, for the following reasons:

1. The integration provides a way to perform a checkpoint, but does not provide a way to resume from checkpoint. Thus if we need to resume, we need to go and read the code of the integration to understand how the checkpoint was made, in order to work out a way to resume the model. If in the future revisions the internal code is changed, then user code might yields incorrect result.

2. The checkpoint callback object has some internal counters (`self._checkpoint._cp_count` and `self._checkpoint._counter`) which are used to decide when to checkpoint and where to checkpoint to. This works fine if the trial is executed fully without interruption. However, if the scheduler/algorithm ever decides to pause a trial and later resume it to continue, then all these counters will be reset (since execution happening at a worker, and that after resuming the trainable function is executed from scratch without any previous memory). As the results, checkpointing mechanism will be messed up, with arbitrary overwriting of checkpoints on each others, rendering them rather harmful if to be relied upon.

As a matter of fact, I had to extend this callback and added quite a few things (a resume method, and additional parameters saved to the checkpoint) in order to resolve the above shortcomings.

### Versions / Dependencies

ray=2.1.0

### Reproduction script

I will not provide any script because the problem is rather straightforward.

### Issue Severity

Low: It annoys or frustrates me.

**Comment by gjoliver** [comment]
@hvavhp, good analysis. why don't you open a PR and upstream your enhancements 😄 ?

**Comment by matthewdeng** [comment]
Closing this since we now expose a `ReportCheckpointCallback` which relies more natively on Keras primitives
